File: backend/semgrep_rules/tests/python/kisa_sw_19_improper_authorization.py
def unsafe(request, Project):
    # ruleid: kisa.sw19.python.external-id-object-access-review
    return Project.objects.get(id=request.GET.get("id"))

def safe(request, Project):
    # ok: kisa.sw19.python.external-id-object-access-review
    return Project.objects.get(id=request.GET.get("id"), owner=request.user)

File: backend/semgrep_rules/tests/javascript/kisa_sw_19_improper_authorization.js
async function unsafe(req, Project) {
  // ruleid: kisa.sw19.javascript.external-id-object-access-review
  return Project.findById(req.params.id);
}

async function safe(req, Project) {
  // ok: kisa.sw19.javascript.external-id-object-access-review
  return Project.findOne({ _id: req.params.id, owner: req.user.id });
}
